curriculum/challenges/english/20-upcoming-python/learn-classes-and-objects-by-building-a-sudoku-solver/6568c0a5edddc3daa65d20b2.md

```diff
@@ -0,0 +1,78 @@
+---
+id: 6568c0a5edddc3daa65d20b2
+title: Step 40
+challengeType: 20
+dashedName: step-40
+---
+
```

Comment:
This step does not work in my opinion. `row` is not defined yet and it's confusing.
We should first create the whole generator expression:
` self.board[row][col] != num for row in range(9)`